frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -304,17 +304,21 @@ export const experimentLogic = kea<experimentLogicType>({
                 trendsLogic.findMounted({ dashboardItemId: values.experimentInsightId })?.actions.setFilters(filters)
             }
         },
+        loadExperiment: async () => {
+            actions.loadExperimentResults('new')
+        },
         loadExperimentSuccess: async ({ experimentData }) => {
             experimentData && eventUsageLogic.actions.reportExperimentViewed(experimentData)
             actions.setExperimentInsightType(experimentData?.filters.insight || InsightType.FUNNELS)
+            actions.loadExperimentResults('new')
             if (!experimentData?.start_date) {
                 // loading a draft experiment
                 actions.setNewExperimentData({ ...experimentData })
                 actions.createNewExperimentInsight(experimentData?.filters)
             } else {
                 actions.resetNewExperiment()
-                actions.loadExperimentResults()
-                actions.loadSecondaryMetricResults()
+                actions.loadExperimentResults(values.experimentId || 'new')
+                actions.loadSecondaryMetricResults(values.experimentId || 'new')
```

Comment:
This is the same architectural problem that is a headache in `insightLogic` – one logic handles both collection-level concerns (such as creating new experiments) and item-level concerns (such as loading the experiment/doing operations on it). Because of this, we have to do this very careful dance with setting the right item ID at all times.
This could be avoided though if we the logic was _keyed_ on item ID. That way experiment ID would be just a prop. `cohortLogic` for instance more-or-less does this.

Reply:
Indeed, set a key on `experimentLogic` to separate it from any other experiment.